slycot/transform.py

```diff
@@ -1236,4 +1236,245 @@ def tg01fd(l,n,m,p,A,E,B,C,Q=None,Z=None,compq='N',compz='N',joba='N',tol=0.0,ld
+            LDWORK >= 1, if MIN(L,N) = 0; otherwise,
+            LDWORK >= MAX( N+P, MIN(L,N)+MAX(3*N-1,M,L) ).
+            If LDWORK >= 2*L*N+L*M+N*P+
+                        MAX( 1, N+P, MIN(L,N)+MAX(3*N-1,M,L) ) then
+            the original matrices are restored if no order reduction
+            is possible. This is achieved by saving system matrices
+            before reduction and restoring them if no order reduction
+            took place.
+
+            If LDWORK = -1, then a workspace query is assumed; the
+            routine only calculates the optimal size of the DWORK
+            array, returns this value as the first entry of the DWORK
+            array, and no error message related to LDWORK is issued by
+            XERBLA. The optimal size does not necessarily include the 
+            space needed for saving the original system matrices.
+
+    Error Indicator
+
+    INFO    INTEGER
+            = 0:  successful exit;
+            < 0:  if INFO = -i, the i-th argument had an illegal
+                    value.
+
+    Method
+
+    The subroutine elliminates the non-dynamics modes in two steps:
+
+    Step 1: Reduce the system to the SVD-like coordinate form
+    (Q'*A*Z-lambda*Q'*E*Z, Q'*B, C*Z) , where
+
+            ( A11 A12 A13 )           ( E11 0 0 )         ( B1 )
+    Q'*A*Z = ( A21 A22  0  ), Q'*E*Z = (  0  0 0 ), Q'*B = ( B2 ),
+            ( A31  0   0  )           (  0  0 0 )         ( B3 )
+
+        C*Z = ( C1  C2  C3 ),
+
+    where E11 and A22 are upper triangular invertible matrices.
+
+    Step 2: Compute the reduced system as (Ar-lambda*Er,Br,Cr,Dr),
+    where
+        ( A11 - A12*inv(A22)*A21, A13 )        ( E11 0 )
+    Ar = (                             ),  Er = (       ),
+        (     A31                  0  )        (  0  0 )
+
+        ( B1 - A12*inv(A22)*B2 )
+    Br = (                      ),  Cr = ( C1 - C2*inv(A22)*A21, C3 ),
+        (        B3            )
+
+    Dr = D - C2*inv(A22)*B2.
+
+    Step 3: If desired (JOBS = 'S'), reduce the descriptor system to
+    the standard form
+
+    Ar <- diag(inv(Er),I)*Ar;  Br <- diag(inv(Er),I)*Br;
+    Er  = diag(I,0).
+
+    If L = N and LR = NR = RANKE, then if Step 3 is performed,
+    the resulting system is a standard state space system.
+
+    Numerical Aspects
+
+    If L = N, the algorithm requires 0( N**3 ) floating point
+    operations.
+
+    """
+
+    hidden = ' (hidden by the wrapper)'
+    arg_list = [
+        'jobs',
+        'l', 'n', 'm', 'p',
+        'A', 'lda'+hidden,
+        'E', 'lde'+hidden,
+        'B', 'ldb'+hidden,
+        'C', 'ldc'+hidden,
+        'D', 'ldd'+hidden,
+        'tol',
+        'dwork'+hidden,
+        'info'
+    ]
+
+    out = _wrapper.tg01gd(jobs,l,n,m,p,a,e,b,c,d,tol)
+    info = out[-1]
+    raise_if_slycot_error(info, arg_list)
+    A,E,B,C,D,lr,nr,ranke,infred,info = out
+    return (
+        A[..., :nr, :nr],
+        E[..., :nr, :nr],
+        B[..., :nr, :],
+        C[..., :, :nr],
```

Comment:
Remove the leading `...` parts. The wrapper only returns two-dimensional arrays.